sonic-chassisd/scripts/chassisd

```diff
@@ -566,6 +618,410 @@ class ModuleUpdater(logger.Logger):
+                if prev_status != str(ModuleBase.MODULE_STATUS_OFFLINE) and current_status == str(ModuleBase.MODULE_STATUS_OFFLINE):
+                    self.log_notice("{} operational status transitioning to offline".format(key))
+
+                    # Persist dpu down time
+                    self.persist_dpu_reboot_time(key)
+
+                elif prev_status == str(ModuleBase.MODULE_STATUS_OFFLINE) and current_status != str(ModuleBase.MODULE_STATUS_OFFLINE):
+                    self.log_notice("{} operational status transitioning to online".format(key))
+                    reboot_cause = try_get(self.chassis.get_module(module_index).get_reboot_cause)
+
+                    if not self.retrieve_dpu_reboot_time(key) is None or self._is_first_boot(key):
+                        # persist reboot cause
+                        self.persist_dpu_reboot_cause(reboot_cause, key)
+                        # publish reboot cause to db
+                        self.update_dpu_reboot_cause_to_db(key)
+
+    def _get_module_info(self, module_index):
+        """
+        Retrieves module info of this module
+        """
+        module_info_dict = {}
+        module_info_dict = dict.fromkeys(self.info_dict_keys, 'N/A')
+        name = try_get(self.chassis.get_module(module_index).get_name)
+        desc = try_get(self.chassis.get_module(module_index).get_description)
+        status = try_get(self.chassis.get_module(module_index).get_oper_status,
+                         default=ModuleBase.MODULE_STATUS_OFFLINE)
+        asics = try_get(self.chassis.get_module(module_index).get_all_asics,
+                        default=[])
+        serial = try_get(self.chassis.get_module(module_index).get_serial)
+
+        module_info_dict[CHASSIS_MODULE_INFO_NAME_FIELD] = name
+        module_info_dict[CHASSIS_MODULE_INFO_DESC_FIELD] = str(desc)
+        module_info_dict[CHASSIS_MODULE_INFO_SLOT_FIELD] = 'N/A'
+        module_info_dict[CHASSIS_MODULE_INFO_OPERSTATUS_FIELD] = str(status)
+        module_info_dict[CHASSIS_MODULE_INFO_SERIAL_FIELD] = str(serial)
+
+        return module_info_dict
+
+
+    def update_dpu_state(self, key, state):
+        """
+        Update DPU state in chassisStateDB using the given key.
+        """
+        try:
+            # Connect to the CHASSIS_STATE_DB using daemon_base
+            if not self.chassis_state_db:
+                self.chassis_state_db = daemon_base.db_connect("CHASSIS_STATE_DB")
+
+            # Fetch the current data for the given key and convert it to a dict
+            current_data = self._convert_to_dict(self.chassis_state_db.hgetall(key))
+
+            if current_data:
+                self.chassis_state_db.delete(key)
+
+            # Prepare the updated data
+            updates = {
+                "dpu_midplane_link_state": state,
+                "dpu_midplane_link_reason": "",
+                "dpu_midplane_link_time": datetime.now().strftime("%Y%m%d %H:%M:%S"),
```

Comment:
@rameshraghupathy  I mean this line 
```suggestion
                "dpu_midplane_link_time": datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
```
Will be displayed as such when we do `show system-health dpu`:
`DPU2    Online         dpu_midplane_link_state  up             2024-11-25 23:38:21`

Reply:
@gpunathilell done